gpflow/models/model.py

```diff
@@ -118,16 +119,43 @@ def predict_f_samples(
     ) -> tf.Tensor:
         """
         Produce samples from the posterior latent function(s) at the input points.
+
+        :param Xnew: DataPoint
+            Input locations at which to draw samples
+        :param num_samples: int
+            Number of samples to draw.
+            If `None` a one sample is drawn and the return shape is [..., N, P],
+            for any other positive integer the return shape contains an extra batch
+            dimension, [..., S, N, P], with S = num_samples.
```

Comment:
how about making this consistent with tensorflow_probability ? would that be possible?

Reply:
These shapes follow the conventions of our multi-output framework, which closely matches tfp.